### ISSUE TITLE
Contributed command failed in new test API

### ISSUE THREAD

**Issue by testforstephen** [microsoft/vscode-java-test#1227]
Failure in test explorer:
![image](https://user-images.githubusercontent.com/14052197/123597266-b3fca280-d825-11eb-84df-cd5a18b65f72.png)

Logs:
```
!ENTRY org.eclipse.jdt.ls.core 1 0 2021-06-28 15:24:36.674
!MESSAGE >> workspace/executeCommand vscode.java.test.findTestPackagesAndTypes

!ENTRY org.eclipse.jdt.ls.core 4 2 2021-06-28 15:24:40.719
!MESSAGE Problems occurred when invoking code from plug-in: "org.eclipse.jdt.ls.core".
!STACK 1
Java Model Exception: Java Model Status [<anonymous #1> {key=Lorg/springframework/samples/petclinic/variables/LambdaScopeTest$625;} [in accept(java.lang.Integer) [in <lambda #1> [in methodInFirstLevel() [in FirstLevel [in LambdaScopeTest [in LambdaScopeTest.java [in org.springframework.samples.petclinic.variables [in src/main/java [in spring-petclinic]]]]]]]]] does not exist]
	at org.eclipse.jdt.internal.core.JavaElement.newNotPresentException(JavaElement.java:577)
	at org.eclipse.jdt.internal.core.JavaElement.openWhenClosed(JavaElement.java:620)
	at org.eclipse.jdt.internal.core.JavaElement.getElementInfo(JavaElement.java:330)
	at org.eclipse.jdt.internal.core.JavaElement.getElementInfo(JavaElement.java:316)
	at org.eclipse.jdt.internal.core.Member.getFlags(Member.java:172)
	at org.eclipse.jdt.internal.junit.util.CoreTestSearchEngine.isAccessibleClass(CoreTestSearchEngine.java:64)
	at org.eclipse.jdt.internal.junit.launcher.JUnit5TestFinder.internalIsTest(JUnit5TestFinder.java:255)
	at org.eclipse.jdt.internal.junit.launcher.JUnit5TestFinder.findTestsInContainer(JUnit5TestFinder.java:222)
	at com.microsoft.java.test.plugin.searcher.JUnit5TestSearcher.findTestItemsInContainer(JUnit5TestSearcher.java:130)
	at com.microsoft.java.test.plugin.util.TestSearchUtils.findTestPackagesAndTypes(TestSearchUtils.java:128)
	at com.microsoft.java.test.plugin.handler.TestDelegateCommandHandler.executeCommand(TestDelegateCommandHandler.java:54)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler$1.run(WorkspaceExecuteCommandHandler.java:215)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler.executeCommand(WorkspaceExecuteCommandHandler.java:205)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$3(JDTLanguageServer.java:502)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:75)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
!SUBENTRY 1 org.eclipse.jdt.core 4 969 2021-06-28 15:24:40.721
!MESSAGE <anonymous #1> {key=Lorg/springframework/samples/petclinic/variables/LambdaScopeTest$625;} [in accept(java.lang.Integer) [in <lambda #1> [in methodInFirstLevel() [in FirstLevel [in LambdaScopeTest [in LambdaScopeTest.java [in org.springframework.samples.petclinic.variables [in src/main/java [in spring-petclinic]]]]]]]]] does not exist
!SUBENTRY 1 org.eclipse.jdt.core 4 969 2021-06-28 15:24:40.721
!MESSAGE <anonymous #1> {key=Lorg/springframework/samples/petclinic/variables/LambdaScopeTest$625;} [in accept(java.lang.Integer) [in <lambda #1> [in methodInFirstLevel() [in FirstLevel [in LambdaScopeTest [in LambdaScopeTest.java [in org.springframework.samples.petclinic.variables [in src/main/java [in spring-petclinic]]]]]]]]] does not exist

!ENTRY org.eclipse.jdt.ls.core 4 0 2021-06-28 15:24:40.725
!MESSAGE Error in calling delegate command handler
!STACK 1
Java Model Exception: Java Model Status [<anonymous #1> {key=Lorg/springframework/samples/petclinic/variables/LambdaScopeTest$625;} [in accept(java.lang.Integer) [in <lambda #1> [in methodInFirstLevel() [in FirstLevel [in LambdaScopeTest [in LambdaScopeTest.java [in org.springframework.samples.petclinic.variables [in src/main/java [in spring-petclinic]]]]]]]]] does not exist]
	at org.eclipse.jdt.internal.core.JavaElement.newNotPresentException(JavaElement.java:577)
	at org.eclipse.jdt.internal.core.JavaElement.openWhenClosed(JavaElement.java:620)
	at org.eclipse.jdt.internal.core.JavaElement.getElementInfo(JavaElement.java:330)
	at org.eclipse.jdt.internal.core.JavaElement.getElementInfo(JavaElement.java:316)
	at org.eclipse.jdt.internal.core.Member.getFlags(Member.java:172)
	at org.eclipse.jdt.internal.junit.util.CoreTestSearchEngine.isAccessibleClass(CoreTestSearchEngine.java:64)
	at org.eclipse.jdt.internal.junit.launcher.JUnit5TestFinder.internalIsTest(JUnit5TestFinder.java:255)
	at org.eclipse.jdt.internal.junit.launcher.JUnit5TestFinder.findTestsInContainer(JUnit5TestFinder.java:222)
	at com.microsoft.java.test.plugin.searcher.JUnit5TestSearcher.findTestItemsInContainer(JUnit5TestSearcher.java:130)
	at com.microsoft.java.test.plugin.util.TestSearchUtils.findTestPackagesAndTypes(TestSearchUtils.java:128)
	at com.microsoft.java.test.plugin.handler.TestDelegateCommandHandler.executeCommand(TestDelegateCommandHandler.java:54)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler$1.run(WorkspaceExecuteCommandHandler.java:215)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler.executeCommand(WorkspaceExecuteCommandHandler.java:205)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$3(JDTLanguageServer.java:502)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:75)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
!SUBENTRY 1 org.eclipse.jdt.core 4 969 2021-06-28 15:24:40.725
!MESSAGE <anonymous #1> {key=Lorg/springframework/samples/petclinic/variables/LambdaScopeTest$625;} [in accept(java.lang.Integer) [in <lambda #1> [in methodInFirstLevel() [in FirstLevel [in LambdaScopeTest [in LambdaScopeTest.java [in org.springframework.samples.petclinic.variables [in src/main/java [in spring-petclinic]]]]]]]]] does not exist
```

**Comment by jdneo** [comment]
What's the content of LambdaScopeTest.java?

**Comment by testforstephen** [comment]
src/main/java/LambdaScopeTest.java

[LambdaScopeTest.java.txt](https://github.com/microsoft/vscode-java-test/files/6724392/LambdaScopeTest.java.txt)

**Comment by jdneo** [comment]
Looks really strange, Both the extension and Eclipse use the same way to discover test but Eclipse works fine.

Meanwhile, I found two ways to avoid this exception.

1. If the inner most `new Runnable` block is commented, no exception will be observed.
2. Or, If open the file `LambdaScopeTest.java` before searching the test items.

**Comment by jdneo** [comment]
Have no idea when the issue has been fixed but I just tried it in `VS Code@1.69.0-insider` + `vscode-java@1.7.0` + `vscode-java-test@0.35.2022061102`, the issue disappears.
